build/Rulesets/Roslyn.ruleset

```diff
@@ -144,4 +144,9 @@
     <Rule Id="IDE1006" Action="Error" />                      <!-- Naming styles                                Task Open()                                         Task OpenAsync()                      -->
     <Rule Id="IDE1006WithoutSuggestion" Action="Info" />
   </Rules>
+  <Rules AnalyzerId="Microsoft.VisualStudio.Threading.Analyzers" RuleNamespace="Microsoft.VisualStudio.Threading.Analyzers">
+    <Rule Id="VSTHRD200" Action="None" />                       <!-- Naming styles                              Task Open()                                         Task OpenAsync()                      -->
+    <Rule Id="VSTHRD010" Action="None" />                       <!-- Visual Studio service should be used on main thread explicitly.-->
```

Comment:
I'd love to see you get this one enabled sometime. It's limited to just the scope of one function (it can't see whether a method always is called on the UI thread to begin with) but even so, you can always add a new caller to your method so a method-level defense system is arguably appropriate anyway.